src/test/java/codeu/controller/ProfileServletTest.java

```diff
@@ -0,0 +1,112 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import codeu.model.data.Profile;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.basic.ProfileStore;
+import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.MessageStore;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class ProfileServletTest {
+
+  private ConversationServlet conversationServlet;
+  private HttpServletRequest mockRequest;
+  private HttpSession mockSession;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+  private ConversationStore mockConversationStore;
+  private UserStore mockUserStore;
+
+  @Before
+  public void setup() {
+    profileServlet = new ProfileServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/profile.jsp"))
+        .thenReturn(mockRequestDispatcher);
+
+    mockProfileStore = Mockito.mock(ProfileStore.class);
+    ProfileServlet.setProfileStore(mockProfileStore);
+
+    mockMessageStore = Mockito.mock(MessageStore.class);
+    MessageServlet.setMessageStore(mockMessageStore);
+
+    mockUserStore = Mockito.mock(UserStore.class);
+    conversationServlet.setUserStore(mockUserStore);
+  }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(UUID.randomUUID(), UUID.randomUUID(), UUID.randomUUID(), "test_message",  Instant.now()));
+    Mockito.when(mockMessageStore.getAllMessages()).thenReturn(fakeMessageList);
+    String fakeUsername = requestUrl.substring("/profile/".length());
+    UUID fakeID = userStore.getUser(username).getId()
+    Profile fakeProfile = profileStore.getProfile(id);
+
+    profileServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("profile", "messages", fakeProfile, fakeMessageList);
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  @Test
+  public void testDoPost_UserNotLoggedIn() throws IOException, ServletException {
```

Comment:
For user not logged in case, I just check if "request.setAttribute("error", "Invalid User") and "request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response)" ran.